dask_planner/src/sql/logical/aggregate.rs

```diff
@@ -67,8 +68,21 @@ impl PyAggregate {
         }
     }
 
-    #[pyo3(name = "isDistinct")]
-    pub fn distinct(&self) -> PyResult<bool> {
+    #[pyo3(name = "isAggExprDistinct")]
+    pub fn distinct_agg_expr(&self, expr: PyExpr) -> PyResult<bool> {
+        Ok(match expr.expr {
+            Expr::AggregateFunction {
+                fun: _,
+                args: _,
+                distinct,
+                filter: _
+            } => distinct,
+            _ => panic!("Encountered a non Aggregate type in agg_func_name"),
```

Comment:
We could raise a py_type_err instead of panic here.

Reply:
Fixed